--- gcc-4.3.1/gcc/config/i386/mingw32.h.orig	2007-08-02 12:49:31.000000000 +0200
+++ gcc-4.3.1/gcc/config/i386/mingw32.h	2008-06-20 14:14:36.000000000 +0200
@@ -50,11 +50,7 @@
 /* Override the standard choice of /usr/include as the default prefix
    to try when searching for header files.  */
 #undef STANDARD_INCLUDE_DIR
-#if TARGET_64BIT_DEFAULT
-#define STANDARD_INCLUDE_DIR "/mingw/include64"
-#else
 #define STANDARD_INCLUDE_DIR "/mingw/include"
-#endif
 #undef STANDARD_INCLUDE_COMPONENT
 #define STANDARD_INCLUDE_COMPONENT "MINGW"
 
