--- gcc-4.7.0/gcc/config/i386/mingw32.h.orig	2012-02-13 22:46:38.000000000 +0100
+++ gcc-4.7.0/gcc/config/i386/mingw32.h	2012-05-08 22:50:29.438704599 +0200
@@ -144,7 +144,7 @@
 
 /* Override startfile prefix defaults.  */
 #ifndef STANDARD_STARTFILE_PREFIX_1
-#define STANDARD_STARTFILE_PREFIX_1 "/mingw/lib/"
+#define STANDARD_STARTFILE_PREFIX_1 "/lib/"
 #endif
 #ifndef STANDARD_STARTFILE_PREFIX_2
 #define STANDARD_STARTFILE_PREFIX_2 ""
--- gcc-4.7.0/gcc/config.gcc.orig	2012-05-08 22:38:05.000000000 +0200
+++ gcc-4.7.0/gcc/config.gcc	2012-05-08 22:48:39.276921083 +0200
@@ -1465,7 +1465,7 @@
 			tmake_file="${tmake_file} i386/t-mingw-w32"
 			;;
 	esac
-        native_system_header_dir=/mingw/include
+        native_system_header_dir=/include
 	target_gtfiles="\$(srcdir)/config/i386/winnt.c"
 	extra_options="${extra_options} i386/cygming.opt i386/mingw.opt"
 	case ${target} in
