--- gcc-6.5.0/libstdc++-v3/src/c++98/ios_failure.cc.orig	2018-08-08 17:40:11.000000000 +0200
+++ gcc-6.5.0/libstdc++-v3/src/c++98/ios_failure.cc	2018-12-09 18:36:29.106481145 +0100
@@ -122,13 +122,16 @@
     // Otherwise proceeed as normal to see if the handler matches.
     return __class_type_info::__do_upcast(dst_type, obj_ptr);
   }
-#else // _GLIBCXX_USE_DUAL_ABI && __cpp_rtti
-  using __iosfailure = ios::failure;
-#endif
 
   void
   __throw_ios_failure(const char* __s __attribute__((unused)))
   { _GLIBCXX_THROW_OR_ABORT(__iosfailure(_(__s))); }
+#else // _GLIBCXX_USE_DUAL_ABI && __cpp_rtti
+
+  void
+  __throw_ios_failure(const char* __s __attribute__((unused)))
+  { _GLIBCXX_THROW_OR_ABORT(ios::failure(_(__s))); }
+#endif
 
 _GLIBCXX_END_NAMESPACE_VERSION
 } // namespace
